### PR TITLE
Add unsupported warning for user-assigned MI in unsupported environments

### PR DESCRIPTION
**Pull request by billwert**
# Description

Add warnings for use of user-assigned managed identities in unsupported environments, per azure/azure-sdk-for-net#26714. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,8 @@
 ### Features Added
 
 ### Breaking Changes
-Removed `disableAuthoriyValidaionSafetyCheck` for GA, will reintroduce in next beta.
+Removed `disableAuthoriyValidaionSafetyCheck` for GA, will reintroduce in next beta. This is not a breaking change from last GA.
```
Comment (chlowell) [inline sdk/identity/azure-identity/CHANGELOG.md:8]:
```suggestion
Removed `disableAuthorityValidationSafetyCheck` for GA, will reintroduce in next beta. This is not a breaking change from last GA.
```

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
 ### Features Added
 
-### Breaking Changes from 1.5.0-beta.2
-Removed `disableAuthoriyValidaionSafetyCheck` for GA, will reintroduce in next beta.
+### Breaking Changes
+Removed `disableAuthorityValidationSafetyCheck` for GA, will reintroduce in next beta. This is not a breaking change from last GA.
```
Comment (g2vinay) [inline sdk/identity/azure-identity/CHANGELOG.md:8]: Add version Info of last stable release as well.

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 3]
```diff
@@ -4,8 +4,8 @@
 
 ### Features Added
```
Comment (g2vinay) [inline sdk/identity/azure-identity/CHANGELOG.md:5]: Add Release Date of Today.

**Review by g2vinay — APPROVED** [review]

**Pull request merged**